src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -276,12 +300,36 @@ def _make_geoname_value(value: XMLValue) -> dict[str, Any]:
     }
 
 
-def _make_integer_value(value: XMLValue) -> dict[str, Any]:
+def _serialise_integer_prop(
+    prop: XMLProperty, res_bn: BNode, prop_name: URIRef, permissions_lookup: dict[str, Permissions]
+) -> dict[str, Any]:
+    g = _make_integer_prop(prop, res_bn, prop_name, permissions_lookup)
+    return serialise_property(g, str(prop_name))
+
+
+def _make_integer_prop(
+    prop: XMLProperty, res_bn: BNode, prop_name: URIRef, permissions_lookup: dict[str, Permissions]
+) -> Graph:
+    g = Graph()
+    for value in prop.values:
+        single_val_bn = BNode()
+        g.add((res_bn, prop_name, single_val_bn))
+        g += _make_integer_value(value, single_val_bn, permissions_lookup)
```

Comment:
Am I understanding correctly, that the only interger-specific thing in this function is the call to `_make_integer_value()`? In that case, wouldn't it make more sense to have a generalized function `_make_value_prop()` that gets the property name and a list of graphs which represent all the nodes you get from repeatedly calling of `_make_integer_value()`?

In case I'm not making much sense here, here's some pseudo code (which is absolutely not correct like this):

```python
def _serialise_integer_prop(prop, res_bn, prop_name, permissions_lookup):
    bn = BNode()
    vals = [_make_integer_value(v, bn, permissions_lookup) for v in prop.values]
    prop_graph = _make_prop(res_bn, prop_name, vals)
    return serialise_property(prop_graph, str(prop_name))

def _make_property(res_bn, prop_name, vals):
    g = Graph()
    for v in vals:
        g.add((res_bn, prop_name, single_val_bn))
        g += v
```

Like this, you wouldn't need two functions for each value type, but just one.

Reply:
In the code you don't actually use the correct bns, but that is not your point. I mistakenly believed, that you have to actually enumerate all the properties so that the framing worked, in that case, it would be inelegant. In the end when everything is rdflib we won't be framing the properties anymore but the resources, then all this piecemeal code will vanish anyway.
But for some reason (I don't get it entirely why) it works as long as you list one valid property name
![Screenshot 2024-05-15 at 08 35 41](https://github.com/dasch-swiss/dsp-tools/assets/103038637/cc78ca34-956a-49f1-9709-be880e117b25)

I will change this so we make one graph for all the properties and then frame that.

Reply:
> In the code you don't actually use the correct bns, but that is not your point. 

Yes... I realized this while writing the code, that's why I double stressed its incorrectness :)

> I mistakenly believed, that you have to actually enumerate all the properties so that the framing worked, in that case, it would be inelegant. In the end when everything is rdflib we won't be framing the properties anymore but the resources, then all this piecemeal code will vanish anyway. But for some reason (I don't get it entirely why) it works as long as you list one valid property name 
> 
> I will change this so we make one graph for all the properties and then frame that.

👍